### ISSUE TITLE
skaffold dev fails on rolebinding yaml

### ISSUE THREAD

**Issue by cconverse711** [GoogleContainerTools/skaffold#2776]
### Expected behavior
Skaffold dev starts up successfully.
### Actual behavior
```
Generating tags...
Tags generated in 19.155µs
Checking cache...
Cache check complete in 18.709µs
Starting deploy...
kubectl client version: 1.15
Cleaning up...
unable to decode "STDIN": json: cannot unmarshal array into Go value of type unstructured.detector
WARN[0001] deployer cleanup: delete: kubectl delete: exit status 1 
FATA[0001] exiting dev mode because first deploy failed: get images from manifests: reading kubernetes YAML: yaml: unmarshal errors:
  line 1: cannot unmarshal !!seq into map[interface {}]interface {}
```

### Information

- Skaffold version: 0.37
- Operating system: Ubuntu 19.04
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta13
kind: Config
deploy:
  kubectl:
    manifests:
    - service.yaml
    - role-binding.yaml
```
- Contents of service.yaml:
```yaml
apiVersion: v1
kind: Service
metadata:
  # This name uniquely identifies the service
  name: minio-service
spec:
  #type: LoadBalancer
  type: NodePort
  ports:
    - port: 9000
      targetPort: 9000
      protocol: TCP
  selector:
    app: my-app
    tier: s3-storage
```

- Contents of role-binding.yaml:
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: default-admin
  namespace: default
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: admin
subjects:
- kind: ServiceAccount
  name: default
  namespace: default
```
If either manifest is commented out in skaffold.yaml, skaffold dev starts up successfully.
If manifest order is swapped in skaffold.yaml, a different error is output:

```
error parsing STDIN: error converting YAML to JSON: yaml: line 3: did not find expected '-' indicator
WARN[0001] deployer cleanup: delete: kubectl delete: exit status 1 
FATA[0001] exiting dev mode because first deploy failed: get images from manifests: reading kubernetes YAML: yaml: line 3: did not find expected '-' indicator
```

### Steps to reproduce the behavior

1. place provided yaml files in working directory
2. `skaffold dev`

**Comment by balopat** [comment]
can you try indenting the array in your subjects? I think that might be the issue

**Comment by balopat** [comment]
the fix is coming soon in #2778
